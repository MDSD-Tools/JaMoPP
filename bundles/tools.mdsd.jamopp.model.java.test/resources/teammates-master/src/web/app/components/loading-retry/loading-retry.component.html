<div *ngIf="shouldShowRetry; else content" class="text-center">
  <p class="text-muted m-1">{{message}}</p>
  <button (click)="retryEvent.emit()" class="btn btn-primary"><i class="fas fa-redo" aria-hidden="true"></i>&nbsp;Retry</button>
</div>

<ng-template #content>
  <ng-content></ng-content>
</ng-template>
