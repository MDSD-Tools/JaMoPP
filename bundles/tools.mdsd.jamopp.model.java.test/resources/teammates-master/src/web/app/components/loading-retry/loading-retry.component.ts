import { Component, EventEmitter, Input, OnInit, Output } from '@angular/core';

/**
 * Retry button for when loading fails.
 */
@Component({
  selector: 'tm-loading-retry',
  templateUrl: './loading-retry.component.html',
  styleUrls: ['./loading-retry.component.scss'],
})
export class LoadingRetryComponent implements OnInit {

  @Input() message: string = '';

  @Input() shouldShowRetry: boolean = false;

  @Output() retryEvent: EventEmitter<any> = new EventEmitter<any>();

  constructor() { }

  ngOnInit(): void {
  }

}
