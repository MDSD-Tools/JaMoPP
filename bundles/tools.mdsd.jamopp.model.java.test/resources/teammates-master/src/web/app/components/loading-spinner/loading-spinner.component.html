<div class="loading-container">
  <div class="row spinner-border text-primary" role="status"></div>
  <div class="row text-secondary">Loading...</div>
</div>
