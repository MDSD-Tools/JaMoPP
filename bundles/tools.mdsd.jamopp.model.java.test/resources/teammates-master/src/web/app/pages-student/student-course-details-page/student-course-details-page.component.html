<h1 class="page-title text-break" *ngIf="course.courseId">Team Details for {{ course.courseId }}</h1>
<tm-loading-retry [shouldShowRetry]="hasLoadingFailed" [message]="'Failed to load details'" (retryEvent)="retryLoading()">
  <div class="section" *ngIf="course.courseId">
    <div *tmIsLoading="isLoadingCourse || isLoadingInstructor || isLoadingStudent" class="card">
      <div class="card-body fill-plain">
        <ng-container *ngIf="course.courseId">
          <div class="row">
            <div class="col-3 text-right">
              <label>Course ID:</label>
            </div>
            <div class="col-9">
              <span id="course-id">{{ course.courseId }}</span>
            </div>
          </div>
          <div class="row">
            <div class="col-3 text-right">
              <label>Course name:</label>
            </div>
            <div class="col-9">
              <span id="course-name">{{ course.courseName }}</span>
            </div>
          </div>
        </ng-container>
        <ng-container *ngIf="instructorDetails && instructorDetails.length">
          <div class="row">
            <div class="col-3 text-right">
              <label>Instructors:</label>
            </div>
            <div class="col-9" style="margin-bottom: .5rem;" id="instructors">
              <div *ngFor="let instructorDetail of instructorDetails">
                {{ instructorDetail.displayedToStudentsAs }}:
                <a href="mailto:{{ instructorDetail.email }}">{{ instructorDetail.name }} ({{ instructorDetail.email }})</a>
              </div>
            </div>
          </div>
        </ng-container>
        <ng-container *ngIf="student.courseId">
          <div class="row">
            <div class="col-3 text-right">
              <label>Your section:</label>
            </div>
            <div class="col-9">
              <span id="student-section">{{ student.sectionName }}</span>
            </div>
          </div>
          <div class="row">
            <div class="col-3 text-right">
              <label>Your team:</label>
            </div>
            <div class="col-9">
              <span id="student-team">{{ student.teamName }}</span>
            </div>
          </div>
          <div class="row">
            <div class="col-3 text-right">
              <label>Your name:</label>
            </div>
            <div class="col-9">
              <span id="student-name">{{ student.name }}</span>
            </div>
          </div>
          <div class="row">
            <div class="col-3 text-right">
              <label>Your email:</label>
            </div>
            <div class="col-9">
              <span id="student-email">{{ student.email }}</span>
            </div>
          </div>
        </ng-container>
      </div>
    </div>
  </div>
  <div *ngIf="!isLoadingTeammates && teammateProfiles.length" class="card bg-light mb-4">
    <div class="card-body">
      <h5 class="d-inline"><strong> Sort By: </strong></h5>
      <div class="btn-group" data-toggle="buttons">
        <button id="sort-name" class="btn btn-light" [disabled]="isSelectedForSorting(SortBy.RESPONDENT_NAME)" (click)="sortTeammatesBy(SortBy.RESPONDENT_NAME)">Name</button>
        <button class="btn btn-light" [disabled]="isSelectedForSorting(SortBy.RESPONDENT_EMAIL)" (click)="sortTeammatesBy(SortBy.RESPONDENT_EMAIL)">Email</button>
        <button class="btn btn-light" [disabled]="isSelectedForSorting(SortBy.STUDENT_GENDER)" (click)="sortTeammatesBy(SortBy.STUDENT_GENDER)">Gender</button>
        <button class="btn btn-light" [disabled]="isSelectedForSorting(SortBy.INSTITUTION)" (click)="sortTeammatesBy(SortBy.INSTITUTION)">Institution</button>
        <button class="btn btn-light" [disabled]="isSelectedForSorting(SortBy.NATIONALITY)" (click)="sortTeammatesBy(SortBy.NATIONALITY)">Nationality</button>
      </div>
    </div>
  </div>
</tm-loading-retry>

<div class="section" *tmIsLoading="isLoadingTeammates">
  <div class="row top-padded">
    <div class="col-md-2" *ngIf="teammateProfiles.length">
      <strong>Your teammates:</strong>
    </div>
    <div class="col-md-10">
      <table>
        <tbody>
        <tr *ngFor="let teammateProfile of teammateProfiles; let i = index">
          <td class="team-member-photo-cell">
            <img src="{{ teammateProfile.photoUrl }}" class="profile-pic" (error)="setDefaultPic(teammateProfile)">
          </td>
          <td class="team-member-profile-cell" id="teammates-details-{{ i }}">
            <div>
              <label>Name:</label> {{ teammateProfile.name }}
            </div>
            <div>
              <label>Email:</label>&nbsp;<a href="mailto:{{ teammateProfile.email }}">{{ teammateProfile.email }}</a>
            </div>
            <div *ngIf="teammateProfile.gender">
              <label>Gender:</label> {{ teammateProfile.gender | genderFormat }}
            </div>
            <div *ngIf="teammateProfile.nationality">
              <label>Nationality:</label> {{ teammateProfile.nationality }}
            </div>
            <div *ngIf="teammateProfile.institute">
              <label>Institute:</label> {{ teammateProfile.institute }}
            </div>
          </td>
        </tr>
        </tbody>
      </table>
    </div>
  </div>
</div>
