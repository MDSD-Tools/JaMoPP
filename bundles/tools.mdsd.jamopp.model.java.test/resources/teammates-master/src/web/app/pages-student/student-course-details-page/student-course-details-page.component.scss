.page-title {
  margin-bottom: 30px;
}

.section {
  margin-bottom: 40px;
}

.profile-pic {
  max-width: 100%;
  max-height: 150px;
}

.fill-plain {
  background-color: #EAEFF5;
}

label {
  font-weight: bold;
}

.top-padded {
  margin-top: 20px;
}

.team-member-photo-cell {
  padding-bottom: 10px;
}

.team-member-profile-cell {
  padding-left: 30px;
  vertical-align: middle;

  label {
    margin-bottom: 2px;
  }
}
