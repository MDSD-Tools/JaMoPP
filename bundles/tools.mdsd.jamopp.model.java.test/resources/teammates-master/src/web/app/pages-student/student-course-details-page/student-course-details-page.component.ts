import { Component, OnInit } from '@angular/core';
import { ActivatedRoute } from '@angular/router';
import { finalize } from 'rxjs/operators';
import { environment } from '../../../environments/environment';
import { CourseService } from '../../../services/course.service';
import { InstructorService } from '../../../services/instructor.service';
import { StatusMessageService } from '../../../services/status-message.service';
import { StudentProfileService } from '../../../services/student-profile.service';
import { StudentService } from '../../../services/student.service';
import { TableComparatorService } from '../../../services/table-comparator.service';
import { Course, Gender, Instructor, Instructors, JoinState, Student, StudentProfile,
  Students } from '../../../types/api-output';
import { SortBy, SortOrder } from '../../../types/sort-properties';
import { ErrorMessageOutput } from '../../error-message-output';

/**
 * A student profile which also has the profile picture URL
 */
export interface StudentProfileWithPicture extends StudentProfile {
  photoUrl: string;
}

/**
 * Student course details page.
 */
@Component({
  selector: 'tm-student-course-details-page',
  templateUrl: './student-course-details-page.component.html',
  styleUrls: ['./student-course-details-page.component.scss'],
})
export class StudentCourseDetailsPageComponent implements OnInit {
  // enum
  Gender: typeof Gender = Gender;
  SortBy: typeof SortBy = SortBy;
  teammateProfilesSortBy: SortBy = SortBy.NONE;

  // data
  student: Student = {
    email: '',
    courseId: '',
    name: '',
    comments: '',
    joinState: JoinState.NOT_JOINED,
    teamName: '',
    sectionName: '',
  };

  course: Course = {
    courseId: '',
    courseName: '',
    institute: '',
    timeZone: 'UTC',
    creationTimestamp: 0,
    deletionTimestamp: 0,
  };

  courseId: string = '';
  instructorDetails: Instructor[] = [];
  teammateProfiles: StudentProfileWithPicture[] = [];

  isLoadingCourse: boolean = false;
  isLoadingStudent: boolean = false;
  isLoadingInstructor: boolean = false;
  isLoadingTeammates: boolean = false;
  hasLoadingFailed: boolean = false;

  constructor(private tableComparatorService: TableComparatorService,
              private route: ActivatedRoute,
              private instructorService: InstructorService,
              private studentProfileService: StudentProfileService,
              private studentService: StudentService,
              private courseService: CourseService,
              private statusMessageService: StatusMessageService) { }

  /**
   * Fetches relevant data to be displayed on page.
   */
  ngOnInit(): void {
    this.route.queryParams.subscribe((queryParams: any) => {
      this.courseId = queryParams.courseid;
      this.loadStudent(queryParams.courseid);
      this.loadCourse(queryParams.courseid);
      this.loadInstructors(queryParams.courseid);
    });
  }

  /**
   * Loads the course details.
   * @param courseid: id of the course queried
   */
  loadCourse(courseId: string): void {
    this.isLoadingCourse = true;
    this.courseService.getCourseAsStudent(courseId)
        .pipe(finalize(() => this.isLoadingCourse = false))
        .subscribe((course: Course) => {
          this.course = course;
        }, (resp: ErrorMessageOutput) => {
          this.hasLoadingFailed = true;
          this.statusMessageService.showErrorToast(resp.error.message);
        });
  }

  /**
   * Loads the current logged-in student of the course.
   * @param courseid: id of the course queried
   */
  loadStudent(courseId: string): void {
    this.isLoadingStudent = true;
    this.studentService.getStudent(courseId)
        .pipe(finalize(() => this.isLoadingStudent = false))
        .subscribe((student: Student) => {
          this.student = student;
          this.loadTeammates(courseId, student.teamName);
        }, (resp: ErrorMessageOutput) => {
          this.hasLoadingFailed = true;
          this.statusMessageService.showErrorToast(resp.error.message);
        });
  }

  /**
   * Loads the teammates of the current student.
   * @param courseid: id of the course queried
   * @param teamName: team of current student
   */
  loadTeammates(courseId: string, teamName: string): void {
    this.isLoadingTeammates = true;
    this.teammateProfiles = [];
    this.studentService.getStudentsFromCourseAndTeam(courseId, teamName)
      .subscribe((students: Students) => {
        // No teammates
        if (students.students.length === 1 && students.students[0].email === this.student.email) {
          this.isLoadingTeammates = false;
        }
        students.students.forEach((student: Student) => {
          // filter away current user
          if (student.email === this.student.email) {
            return;
          }

          this.studentProfileService.getStudentProfile(student.email, courseId)
                .pipe(finalize(() => this.isLoadingTeammates = false))
                .subscribe((studentProfile: StudentProfile) => {
                  const newPhotoUrl: string =
                    `${environment.backendUrl}/webapi/student/profilePic`
                    + `?courseid=${courseId}&studentemail=${student.email}`;

                  const newTeammateProfile: StudentProfileWithPicture = {
                    ...studentProfile,
                    email: student.email,
                    name: student.name,
                    photoUrl : newPhotoUrl,
                  };

                  this.teammateProfiles.push(newTeammateProfile);
                }, (resp: ErrorMessageOutput) => {
                  this.hasLoadingFailed = true;
                  this.statusMessageService.showErrorToast(resp.error.message);
                });
        });
      }, (resp: ErrorMessageOutput) => {
        this.isLoadingTeammates = false;
        this.hasLoadingFailed = true;
        this.statusMessageService.showErrorToast(resp.error.message);
      });
  }

  /**
   * Loads the instructors of the course.
   * @param courseid: id of the course queried
   */
  loadInstructors(courseId: string): void {
    this.isLoadingInstructor = true;
    this.instructorService.loadInstructors({ courseId })
        .pipe(finalize(() => this.isLoadingInstructor = false))
        .subscribe((instructors: Instructors) => {
          this.instructorDetails = instructors.instructors;
        }, (resp: ErrorMessageOutput) => {
          this.hasLoadingFailed = true;
          this.statusMessageService.showErrorToast(resp.error.message);
        });
  }

  /**
   * Sets the profile picture of a student as the default image
   */
  setDefaultPic(teammateProfile: StudentProfileWithPicture): void {
    teammateProfile.photoUrl = '/assets/images/profile_picture_default.png';
  }

  /**
   * Checks the option selected to sort teammates.
   * @param sortOption: option for sorting
   */
  isSelectedForSorting(sortOption: SortBy): boolean {
    return this.teammateProfilesSortBy === sortOption;
  }

  /**
   * Sorts the teammates according to selected option.
   * @param sortOption: option for sorting
   */
  sortTeammatesBy(sortOption: SortBy): void {
    this.teammateProfilesSortBy = sortOption;

    if (this.teammateProfiles.length > 1) {
      this.teammateProfiles.sort(this.sortPanelsBy(sortOption));
    }
  }

  /**
   * Sorts the panels of teammates in order.
   * @param sortOption: option for sorting
   */
  sortPanelsBy(sortOption: SortBy):
      ((a: StudentProfile, b: StudentProfile) => number) {
    return ((a: StudentProfile, b: StudentProfile): number => {
      let strA: string;
      let strB: string;
      switch (sortOption) {
        case SortBy.RESPONDENT_NAME:
          strA = a.name;
          strB = b.name;
          break;
        case SortBy.RESPONDENT_EMAIL:
          strA = a.email;
          strB = b.email;
          break;
        case SortBy.STUDENT_GENDER:
          strA = a.gender;
          strB = b.gender;
          break;
        case SortBy.INSTITUTION:
          strA = a.institute;
          strB = b.institute;
          break;
        case SortBy.NATIONALITY:
          strA = a.nationality;
          strB = b.nationality;
          break;
        default:
          strA = '';
          strB = '';
      }
      return this.tableComparatorService.compare(sortOption, SortOrder.ASC, strA, strB);
    });
  }

  retryLoading(): void {
    this.hasLoadingFailed = false;
    this.loadCourse(this.courseId);
    this.loadInstructors(this.courseId);
    this.loadStudent(this.courseId);
  }
}
