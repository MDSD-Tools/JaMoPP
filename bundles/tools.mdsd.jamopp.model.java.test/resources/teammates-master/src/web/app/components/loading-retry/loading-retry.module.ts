import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { LoadingRetryComponent } from './loading-retry.component';

/**
 * Module for the retry button
 */
@NgModule({
  declarations: [LoadingRetryComponent],
  imports: [
    CommonModule,
  ],
  exports: [LoadingRetryComponent],
})
export class LoadingRetryModule { }
