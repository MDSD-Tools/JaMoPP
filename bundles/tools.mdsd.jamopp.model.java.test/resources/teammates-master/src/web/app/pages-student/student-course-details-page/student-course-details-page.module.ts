import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { RouterModule, Routes } from '@angular/router';
import { LoadingRetryModule } from '../../components/loading-retry/loading-retry.module';
import { LoadingSpinnerModule } from '../../components/loading-spinner/loading-spinner.module';
import { TeammatesCommonModule } from '../../components/teammates-common/teammates-common.module';
import { StudentCourseDetailsPageComponent } from './student-course-details-page.component';

const routes: Routes = [
  {
    path: '',
    component: StudentCourseDetailsPageComponent,
  },
];

/**
 * Module for student course details page.
 */
@NgModule({
  declarations: [
    StudentCourseDetailsPageComponent,
  ],
  exports: [
    StudentCourseDetailsPageComponent,
  ],
  imports: [
    CommonModule,
    RouterModule.forChild(routes),
    TeammatesCommonModule,
    LoadingSpinnerModule,
    LoadingRetryModule,
  ],
})
export class StudentCourseDetailsPageModule { }
