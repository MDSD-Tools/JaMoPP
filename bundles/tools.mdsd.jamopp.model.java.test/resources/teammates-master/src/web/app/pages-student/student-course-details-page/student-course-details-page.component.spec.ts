import { HttpClientTestingModule } from '@angular/common/http/testing';
import { async, ComponentFixture, TestBed } from '@angular/core/testing';
import { RouterTestingModule } from '@angular/router/testing';
import { Course, Gender, Instructor, InstructorPermissionRole, JoinState, Student } from '../../../types/api-output';
import { LoadingRetryModule } from '../../components/loading-retry/loading-retry.module';
import { LoadingSpinnerModule } from '../../components/loading-spinner/loading-spinner.module';
import { TeammatesCommonModule } from '../../components/teammates-common/teammates-common.module';
import { StudentCourseDetailsPageComponent, StudentProfileWithPicture } from './student-course-details-page.component';

describe('StudentCourseDetailsPageComponent', () => {
  let component: StudentCourseDetailsPageComponent;
  let fixture: ComponentFixture<StudentCourseDetailsPageComponent>;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      declarations: [StudentCourseDetailsPageComponent],
      imports: [
        HttpClientTestingModule,
        RouterTestingModule,
        TeammatesCommonModule,
        LoadingSpinnerModule,
        LoadingRetryModule,
      ],
    })
    .compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(StudentCourseDetailsPageComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('should snap with default fields', () => {
    expect(fixture).toMatchSnapshot();
  });

  it('should snap with populated fields', () => {
    const student: Student = {
      courseId: '1.1.c-demo2',
      email: '1@1.com',
      name: '1',
      comments: '',
      joinState: JoinState.NOT_JOINED,
      sectionName: 'Tutorial Group 2',
      teamName: 'Team 2',
    };

    const instructorDetails: Instructor[] = [{
      googleId: '',
      courseId: '1.1.c-demo2',
      displayedToStudentsAs: 'Instructor',
      isDisplayedToStudents: true,
      email: '1@1.com',
      name: '1',
      joinState: JoinState.JOINED,
      role: InstructorPermissionRole.INSTRUCTOR_PERMISSION_ROLE_COOWNER,
    }];

    const course: Course = {
      courseId: '1.1.c-demo2',
      courseName: 'Sample Course 101',
      institute: 'Test Institute',
      creationTimestamp: 1552472130000,
      deletionTimestamp: 0,
      timeZone: 'UTC',
    };

    const teammateProfiles: StudentProfileWithPicture[] = [
      {
        photoUrl: '/assets/images/profile_picture_default.png',
        email: 'iam2@hello.com',
        gender: Gender.MALE,
        institute: 'nus',
        moreInfo: 'Misc',
        name: '2',
        nationality: 'Andorran',
        shortName: 'I am 2',
      },
    ];

    component.course = course;
    component.instructorDetails = instructorDetails;
    component.student = student;
    component.teammateProfiles = teammateProfiles;
    component.isLoadingStudent = false;
    component.isLoadingInstructor = false;
    component.isLoadingTeammates = false;
    component.isLoadingCourse = false;

    fixture.detectChanges();
    expect(fixture).toMatchSnapshot();
  });

  it('should snap when all data are still loading', () => {
    component.isLoadingStudent = true;
    component.isLoadingInstructor = true;
    component.isLoadingTeammates = true;
    component.isLoadingCourse = true;

    fixture.detectChanges();
    expect(fixture).toMatchSnapshot();
  });
});
