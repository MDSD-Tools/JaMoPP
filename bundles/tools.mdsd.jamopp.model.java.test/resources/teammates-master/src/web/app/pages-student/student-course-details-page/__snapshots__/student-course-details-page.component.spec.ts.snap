// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StudentCourseDetailsPageComponent should snap when all data are still loading 1`] = `
<tm-student-course-details-page
  Gender={[Function Object]}
  SortBy={[Function Object]}
  course={[Function Object]}
  courseId="undefined"
  courseService={[Function CourseService]}
  hasLoadingFailed="false"
  instructorDetails={[Function Array]}
  instructorService={[Function InstructorService]}
  isLoadingCourse={[Function Boolean]}
  isLoadingInstructor={[Function Boolean]}
  isLoadingStudent={[Function Boolean]}
  isLoadingTeammates={[Function Boolean]}
  route={[Function ActivatedRoute]}
  statusMessageService={[Function StatusMessageService]}
  student={[Function Object]}
  studentProfileService={[Function StudentProfileService]}
  studentService={[Function StudentService]}
  tableComparatorService={[Function TableComparatorService]}
  teammateProfiles={[Function Array]}
  teammateProfilesSortBy="0"
>
  <tm-loading-retry>
    
    
    
    
    
  </tm-loading-retry><tm-loading-spinner>
    <div
      class="loading-container"
    >
      <div
        class="row spinner-border text-primary"
        role="status"
      />
      <div
        class="row text-secondary"
      >
        Loading...
      </div>
    </div>
  </tm-loading-spinner>
</tm-student-course-details-page>
`;

exports[`StudentCourseDetailsPageComponent should snap with default fields 1`] = `
<tm-student-course-details-page
  Gender={[Function Object]}
  SortBy={[Function Object]}
  course={[Function Object]}
  courseId="undefined"
  courseService={[Function CourseService]}
  hasLoadingFailed="false"
  instructorDetails={[Function Array]}
  instructorService={[Function InstructorService]}
  isLoadingCourse={[Function Boolean]}
  isLoadingInstructor={[Function Boolean]}
  isLoadingStudent={[Function Boolean]}
  isLoadingTeammates="false"
  route={[Function ActivatedRoute]}
  statusMessageService={[Function StatusMessageService]}
  student={[Function Object]}
  studentProfileService={[Function StudentProfileService]}
  studentService={[Function StudentService]}
  tableComparatorService={[Function TableComparatorService]}
  teammateProfiles={[Function Array]}
  teammateProfilesSortBy="0"
>
  <tm-loading-retry>
    
    
    
    
    
  </tm-loading-retry><div
    class="section"
  >
    <div
      class="row top-padded"
    >
      
      <div
        class="col-md-10"
      >
        <table>
          <tbody>
            
          </tbody>
        </table>
      </div>
    </div>
  </div>
</tm-student-course-details-page>
`;

exports[`StudentCourseDetailsPageComponent should snap with populated fields 1`] = `
<tm-student-course-details-page
  Gender={[Function Object]}
  SortBy={[Function Object]}
  course={[Function Object]}
  courseId="undefined"
  courseService={[Function CourseService]}
  hasLoadingFailed="false"
  instructorDetails={[Function Array]}
  instructorService={[Function InstructorService]}
  isLoadingCourse="false"
  isLoadingInstructor="false"
  isLoadingStudent="false"
  isLoadingTeammates="false"
  route={[Function ActivatedRoute]}
  statusMessageService={[Function StatusMessageService]}
  student={[Function Object]}
  studentProfileService={[Function StudentProfileService]}
  studentService={[Function StudentService]}
  tableComparatorService={[Function TableComparatorService]}
  teammateProfiles={[Function Array]}
  teammateProfilesSortBy="0"
>
  <h1
    class="page-title text-break"
  >
    Team Details for 1.1.c-demo2
  </h1><tm-loading-retry>
    
    
    <div
      class="section"
    >
      
      <div
        class="card"
      >
        <div
          class="card-body fill-plain"
        >
          
          
          <div
            class="row"
          >
            <div
              class="col-3 text-right"
            >
              <label>
                Course ID:
              </label>
            </div>
            <div
              class="col-9"
            >
              <span
                id="course-id"
              >
                1.1.c-demo2
              </span>
            </div>
          </div>
          <div
            class="row"
          >
            <div
              class="col-3 text-right"
            >
              <label>
                Course name:
              </label>
            </div>
            <div
              class="col-9"
            >
              <span
                id="course-name"
              >
                Sample Course 101
              </span>
            </div>
          </div>
          
          
          <div
            class="row"
          >
            <div
              class="col-3 text-right"
            >
              <label>
                Instructors:
              </label>
            </div>
            <div
              class="col-9"
              id="instructors"
              style="margin-bottom: .5rem;"
            >
              
              <div>
                 Instructor: 
                <a
                  href="mailto:1@1.com"
                >
                  1 (1@1.com)
                </a>
              </div>
            </div>
          </div>
          
          
          <div
            class="row"
          >
            <div
              class="col-3 text-right"
            >
              <label>
                Your section:
              </label>
            </div>
            <div
              class="col-9"
            >
              <span
                id="student-section"
              >
                Tutorial Group 2
              </span>
            </div>
          </div>
          <div
            class="row"
          >
            <div
              class="col-3 text-right"
            >
              <label>
                Your team:
              </label>
            </div>
            <div
              class="col-9"
            >
              <span
                id="student-team"
              >
                Team 2
              </span>
            </div>
          </div>
          <div
            class="row"
          >
            <div
              class="col-3 text-right"
            >
              <label>
                Your name:
              </label>
            </div>
            <div
              class="col-9"
            >
              <span
                id="student-name"
              >
                1
              </span>
            </div>
          </div>
          <div
            class="row"
          >
            <div
              class="col-3 text-right"
            >
              <label>
                Your email:
              </label>
            </div>
            <div
              class="col-9"
            >
              <span
                id="student-email"
              >
                1@1.com
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
    
    <div
      class="card bg-light mb-4"
    >
      <div
        class="card-body"
      >
        <h5
          class="d-inline"
        >
          <strong>
             Sort By: 
          </strong>
        </h5>
        <div
          class="btn-group"
          data-toggle="buttons"
        >
          <button
            class="btn btn-light"
            id="sort-name"
          >
            Name
          </button>
          <button
            class="btn btn-light"
          >
            Email
          </button>
          <button
            class="btn btn-light"
          >
            Gender
          </button>
          <button
            class="btn btn-light"
          >
            Institution
          </button>
          <button
            class="btn btn-light"
          >
            Nationality
          </button>
        </div>
      </div>
    </div>
    
    
  </tm-loading-retry><div
    class="section"
  >
    <div
      class="row top-padded"
    >
      
      <div
        class="col-md-2"
      >
        <strong>
          Your teammates:
        </strong>
      </div>
      <div
        class="col-md-10"
      >
        <table>
          <tbody>
            
            <tr>
              <td
                class="team-member-photo-cell"
              >
                <img
                  class="profile-pic"
                  src="/assets/images/profile_picture_default.png"
                />
              </td>
              <td
                class="team-member-profile-cell"
                id="teammates-details-0"
              >
                <div>
                  <label>
                    Name:
                  </label>
                   2 
                </div>
                <div>
                  <label>
                    Email:
                  </label>
                  Â 
                  <a
                    href="mailto:iam2@hello.com"
                  >
                    iam2@hello.com
                  </a>
                </div>
                
                <div>
                  <label>
                    Gender:
                  </label>
                   Male 
                </div>
                
                <div>
                  <label>
                    Nationality:
                  </label>
                   Andorran 
                </div>
                
                <div>
                  <label>
                    Institute:
                  </label>
                   nus 
                </div>
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  </div>
</tm-student-course-details-page>
`;
